draft-ietf-tls-tls13.md

```diff
@@ -3381,50 +3495,36 @@ The following table describes the inputs to the key calculation for
 each class of traffic keys:
 
 ~~~
-  Record Type Secret  Label                              Handshake Hash
-  ----------- ------  -----                             ---------------
-  Early data     xSS  "early data key expansion"            ClientHello
-
+  Record Type Secret  Label                           Handshake Context
+  ----------- ------  -----                           -----------------
+  0-RTT          xSS  "early handshake                     ClientHello
+  Handshake            key expansion              + ServerConfiguration
+                                                   + Server Certificate  
+
+  0-RTT          xSS  "early application                    ClientHello
+  Application          data key expansion"        + ServerConfiguration
+                                                   + Server Certificate
+                                                   
   Handshake      xES  "handshake key expansion"          ClientHello...
                                                             ServerHello
 
-  Application  master "application data key expansion"    All handshake
-               secret                                      messages but
-                                                               Finished
-                                                         (session_hash)
+  Application  master "application data key expansion"   ClientHello...
+               secret                                   Server Finished
 ~~~
 
 ###  The Handshake Hash
 
 
-       handshake_hash = Hash(
-                             Hash(handshake_messages) ||
-                             Hash(configuration)
-                            )
-
-handshake_messages
-: All handshake messages sent or
-  received, starting at ClientHello up to the present time, with the
-  exception of the Finished message, including the type and length
-  fields of the handshake messages. This is the concatenation of all the
-  exchanged Handshake structures in plaintext form (even if they
-  were encrypted on the wire).
-
-configuration
-: When 0-RTT is in use ({{early-data-indication}})
-this contains the concatenation of the ServerConfiguration and Certificate
-messages from the handshake where the configuration was established (including the
-type and length fields). Note that
-this requires the client and server to memorize these values.
-{:br }
+The handshake hash is defined as the hash of all handshake messages
+sent or received, starting at ClientHello up to the present time,
+with the exception of the Finished message, including the type and
+length fields of the handshake messages. This is the concatenation
+of all the exchanged Handshake structures in plaintext form (even if
+they were encrypted on the wire).
```

Comment:
Need to explicitly exclude post-handshake messages. NewSessionTicket, CertificateRequest, etc...